tests/registry.cc

```diff
@@ -73,52 +129,17 @@ main( int argc, char * argv[] )
 /* -------------------------------------------------------------------------- */
 
   /* Initialize common registry. */
-  flox::from_json( R"( {
-      "inputs": {
-        "nixpkgs": {
-          "from": {
-            "type": "github"
-          , "owner": "NixOS"
-          , "repo": "nixpkgs"
-          , "rev": "e8039594435c68eb4f780f3e9bf3972a7399c4b1"
-          }
-        , "subtrees": ["legacyPackages"]
-        }
-      , "floco": {
-          "from": {
-            "type": "github"
-          , "owner": "aakropotkin"
-          , "repo": "floco"
-          , "rev": "1e84b4b16bba5746e1195fa3a4d8addaaf2d9ef4"
-          }
-        , "subtrees": ["packages"]
-        }
-      , "nixpkgs-flox": {
-          "from": {
-            "type": "github"
-          , "owner": "flox"
-          , "repo": "nixpkgs-flox"
-          , "rev": "feb593b6844a96dd4e17497edaabac009be05709"
-          }
-        , "subtrees": ["catalog"]
-        , "stabilities": ["stable"]
-        }
-      }
-    , "defaults": {
-        "subtrees": null
-      , "stabilities": ["stable"]
-      }
-    , "priority": ["nixpkgs", "floco", "nixpkgs-flox"]
-    } )"_json
-  , commonRegistry
-  );
+  flox::from_json( TEST_DATA_DIR "/registry/registry0.json", commonRegistry );
```

Comment:
```suggestion
```

You need to read the file before passing it to `from_json`.

Here you're saying "convert the string `TEST_DATA_DIR "/registry/registry0.json"` to a `RegistryRaw`, which will throw an error.

You can actually just drop this block though since the other test cases you added already test whether the contents of that file can be converted from JSON.

Reply:
Fixed